src/librustdoc/passes/unindent_comments/tests.rs

```diff
@@ -1,21 +1,40 @@
 use super::*;
 use rustc_span::source_map::DUMMY_SP;
+use rustc_span::symbol::Symbol;
+use rustc_span::with_default_session_globals;
 
 fn create_doc_fragment(s: &str) -> Vec<DocFragment> {
     vec![DocFragment {
         line: 0,
         span: DUMMY_SP,
         parent_module: None,
-        doc: s.to_string(),
+        doc: Symbol::intern(s),
         kind: DocFragmentKind::SugaredDoc,
+        need_backline: false,
+        indent: 0,
     }]
 }
 
 #[track_caller]
 fn run_test(input: &str, expected: &str) {
-    let mut s = create_doc_fragment(input);
-    unindent_fragments(&mut s);
-    assert_eq!(s[0].doc, expected);
+    with_default_session_globals(|| {
+        let mut s = create_doc_fragment(input);
+        unindent_fragments(&mut s);
+        assert_eq!(
+            &s[0]
+                .doc
+                .as_str()
+                .lines()
+                .map(|l| if l.len() > s[0].indent {
+                    l[s[0].indent..].to_string()
+                } else {
+                    String::new()
+                })
+                .collect::<Vec<_>>()
+                .join("\n"),
+            expected
+        );
```

Comment:
This is a little concerning - it's no longer possible to tell what the input is actually being turned into. If there were a bug here, it would be impossible to tell apart from a bug in rustdoc proper; the test could succeed when it shouldn't or spuriously fail. Can you instead change the hard-coded inputs at the callsite to what this transforms them into?

Reply:
I'm a bit lost here: the `expected` is what we want the transformation to turn into. When the `assert_eq!` fails, you see what you got vs what was expected (I can tell because I had it a few times already...).

Reply:
`expected` is what you want the transformation to turn into *after* the long intermediate calculation with indents. I think instead we should be comparing it to just after `doc_fragments`:
```
        let mut s = create_doc_fragment(input);
        unindent_fragments(&mut s);
        assert_eq!(s[0], expected);

Reply:
Otherwise, it's hard to tell that the test is testing the right thing, even if it passes.

Reply:
The unindent itself just compute how much characters we have to skip at the beginning of each line. We don't change the doc fragment string itself, which is why we have to put it together by applying the `indent` field on each line.

Reply:
> We don't change the doc fragment string itself

That's *exactly* my point - you're not testing the fragment string, you're testing some function applied to the fragment string. I want to see what unindent_fragments is doing, not what the test suite is post-processing it into.

Reply:
But what you suggested would simply be comparing the original string with itself. It wouldn't make much sense.

Reply:
As discussed on discord, this snippet isn't unique to the test suite; there's another part of rustdoc with the same code and you copied it here. You're going to refactor it into a function instead so it's clear that the code is the same in both places.